api/types/container/hostconfig.go

```diff
@@ -320,7 +320,11 @@ func ValidateRestartPolicy(policy RestartPolicy) error {
 	switch policy.Name {
 	case RestartPolicyAlways, RestartPolicyUnlessStopped, RestartPolicyDisabled:
 		if policy.MaximumRetryCount != 0 {
-			return &errInvalidParameter{fmt.Errorf("invalid restart policy: maximum retry count cannot be used with restart policy '%s'", policy.Name)}
+			msg := "invalid restart policy: maximum retry count can only be used with 'on-failure'"
+			if policy.MaximumRetryCount < 0 {
+				msg += " and cannot be negative"
+			}
+			return &errInvalidParameter{fmt.Errorf(msg)}
```

Comment:
nit: 
```suggestion
			return &errInvalidParameter{errors.New(msg)}
```

Reply:
Ah, yes; was considering that, but we didn't import `"errors"` yet, so thought I'd keep it at `fmt.Errorf()` 😅

Reply:
Works for me :)